packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -483,6 +523,11 @@ void _validateFlutter(YamlMap yaml, List<String> errors) {
         break;
       case 'generate':
         break;
+      case 'deferred-components':
+        if (kvp.value is! YamlList || kvp.value[0] is! YamlMap) {
```

Comment:
I would move some of the validation into here, checking each map item for the correct type at least (see the existing tests)